src/ExternalImage.php

```diff
@@ -0,0 +1,479 @@
+     * ```twig
+     * {% if post.thumbnail.aspect < 1 %}
+     *   {# handle vertical image #}
+     *   <img src="{{ post.thumbnail.src|resize(300, 500) }}" alt="A basketball player" />
+     * {% else %}
+     *   <img src="{{ post.thumbnail.src|resize(500) }}" alt="A sumo wrestler" />
+     * {% endif %}
+     * ```
+     *
+     * @api
+     */
+    public function aspect()
+    {
+        return $this->imageDimensions->aspect();
+    }
+
+    /**
+     * Inits the object with an absolute path.
+     *
+     * @param string $file_path An absolute path to a file.
+     * @internal
+     *
+     */
+    protected function init_with_file_path($file_path)
+    {
+        $url = URLHelper::file_system_to_url($file_path);
+
+        $this->abs_url = $url;
+        $this->file_loc = $file_path;
+        $this->file = $file_path;
+        $this->imageDimensions = new ImageDimensions($file_path);
+        $this->file_size = new FileSize($file_path);
+    }
+
+    /**
+     * Inits the object with a relative path.
+     *
+     * @param string $relative_path A relative path to a file.
+     * @internal
+     *
+     */
+    protected function init_with_relative_path($relative_path)
+    {
+        $file_path = URLHelper::get_full_path($relative_path);
+
+        $this->abs_url = home_url($relative_path);
+        $this->file_loc = $file_path;
+        $this->file = $file_path;
+        $this->imageDimensions = new ImageDimensions($file_path);
+        $this->file_size = new FileSize($file_path);
+    }
+
+    /**
+     * Inits the object with an URL.
+     *
+     * @param string $url An URL on the same host.
+     * @internal
+     *
+     */
+    protected function init_with_url($url)
+    {
+        $this->abs_url = $url;
+
+        if (URLHelper::is_local($url)) {
+            $this->file = URLHelper::remove_double_slashes(
+                ABSPATH . URLHelper::get_rel_url($url)
+            );
+            $this->file_loc = URLHelper::remove_double_slashes(
+                ABSPATH . URLHelper::get_rel_url($url)
+            );
+            $this->imageDimensions = new ImageDimensions($this->file_loc);
+            $this->file_size = new FileSize($this->file_loc);
+        } else {
+            $this->imageDimensions = new ImageDimensions();
+        }
+    }
+
+    /**
+     * Gets the alt text for an image.
+     *
+     * For better accessibility, you should always add an alt attribute to your images, even if it’s
+     * empty.
+     *
+     * @return string Alt text stored in WordPress.
+     * @example
+     * ```twig
+     * <img src="{{ image.src }}" alt="{{ image.alt }}" />
+     * ```
+     * ```html
+     * <img src="http://example.org/wp-content/uploads/2015/08/pic.jpg"
+     *     alt="You should always add alt texts to your images for better accessibility" />
+     * ```
+     *
+     * @api
+     */
+    public function alt()
+    {
+        return $this->alt_text;
+    }
+
+    public function caption()
+    {
+        return "";
```

Comment:
If we allow users to set an alt text, should we also allow them to set a caption?

Reply:
Yes, we definitely should. Adding a setter.